src/benchmark.cc

```diff
@@ -184,8 +184,10 @@ void State::PauseTiming() {
   BM_CHECK(started_ && !finished_ && !error_occurred_);
   timer_->StopTimer();
   if (perf_counters_measurement_) {
-    auto measurements = perf_counters_measurement_->StopAndGetMeasurements();
-    for (const auto& name_and_measurement : measurements) {
+    internal::PerfCountersMeasurement::Result measurement_result;
```

Comment:
this would be slightly simpler if `Stop` returns the validity of the read:

```c++
internal::PerfCountersMeasurement::Results = measurements;
bool is_valid = perf_counters_measurement_->Stop(&measurements);
BM_CHECK(is_valid) << "Perf counters value read failed";
for (const auto& name_and_measurement : measurements) {
...
}
```

Reply:
If `Stop` returns the validity, do we need to remove `PerfCountersMeasurement::Results` 
and continue to use `std::vector<std::pair<std::string, double>> measurements`?

Reply:
No, just return a bool from `Stop`. It already takes an object to pour in the results. So the code becomes simpler, like @dominichamon  pointed out:

```
if(!perf_counters_measurement_->Stop(&measurement_result))
  BM_CHECK(false) << "Perf counters value read failed"
```
@dominichamon  - is there a more appropriate way to report the error (`std:err << ` and exit?)